turbo/jsonrpc/erigon_system.go

```diff
@@ -66,6 +68,15 @@ func (api *ErigonImpl) BlockNumber(ctx context.Context, rpcBlockNumPtr *rpc.Bloc
 			return 0, err
 		}
 	case rpc.FinalizedBlockNumber:
+		if borwhitelist.GetWhitelistingService() != nil {
```

Comment:
this block should be moved to rpchelper.GetFinalizedBlockNumber

why rawdb.ReadForkchoiceFinalized is not working for bor? is it possible to precalculate this and save there?